pycyphal/transport/serial/_frame.py

```diff
@@ -132,58 +154,72 @@ def parse_from_cobs_image(image: memoryview) -> typing.Optional[SerialFrame]:
         return SerialFrame.parse_from_unescaped_image(memoryview(unescaped_image))
 
     @staticmethod
-    def parse_from_unescaped_image(header_payload_image: memoryview) -> typing.Optional[SerialFrame]:
+    def parse_from_unescaped_image(image: memoryview) -> typing.Optional[SerialFrame]:
         """
         :returns: Frame or None if the image is invalid.
         """
-        if len(header_payload_image) < _HEADER_SIZE:
-            return None
-
-        header = header_payload_image[:_HEADER_SIZE]
-        if not pycyphal.transport.commons.crc.CRC32C.new(header).check_residue():
-            return None
-
-        payload = header_payload_image[_HEADER_SIZE:]
-
-        # noinspection PyTypeChecker
-        (
-            version,
-            int_priority,
-            src_nid,
-            dst_nid,
-            int_data_spec,
-            transfer_id,
-            index_eot,
-        ) = _HEADER_WITHOUT_CRC_FORMAT.unpack_from(header)
-        if version != _VERSION:
+        try:
+            (
+                version,
+                int_priority,
+                source_node_id,
+                destination_node_id,
+                data_specifier_snm,
+                transfer_id,
+                frame_index_eot,
+                user_data,
+            ) = _HEADER_FORMAT_NO_CRC.unpack_from(image)
+        except struct.error:
             return None
 
-        src_nid = None if src_nid == _ANONYMOUS_NODE_ID else src_nid
-        dst_nid = None if dst_nid == _ANONYMOUS_NODE_ID else dst_nid
-
-        try:  # https://github.com/OpenCyphal/pycyphal/issues/176
-            data_specifier: pycyphal.transport.DataSpecifier
-            if int_data_spec & (1 << 15) == 0:
-                data_specifier = pycyphal.transport.MessageDataSpecifier(int_data_spec)
-            else:
-                if int_data_spec & (1 << 14):
-                    role = pycyphal.transport.ServiceDataSpecifier.Role.RESPONSE
+        try:
+            if version == SerialFrame.VERSION:
```

Comment:
This is fine, but I think the commonly accepted practice is to use early return so that you don't have to indent the conditional block so much, as it hurts readability (especially in a language that follows the off-side rule).